### PR TITLE
JEDI/SOCA cap modifications for T,S,U, and V

### PR DESCRIPTION
**Pull request by mark-a-potts**
This PR exposes the full 3d fields for T, S, U, and V to SOCA when built with the UFS in NG-GODAS or S2S mode. 

Several methods and variable have been made public and a StateExport_3d routine has been added.

### CODE REVIEW

**Comment by jiandewang** [comment]
@mark-a-potts : can you combine your commit history into one. The general rule for MOM6 is not to use squash in merging but developers using as less commit history as possible. You can do force push from now on when you need to do further modification. Thanks

File: config_src/drivers/nuopc_cap/mom_cap_methods.F90 [hunk 1]
```diff
@@ -352,14 +354,15 @@ subroutine mom_export(ocean_public, ocean_grid, ocean_state, exportState, clock,
   real                            :: slope, u_min, u_max
   integer                         :: day, secs
   type(ESMF_TimeInterval)         :: timeStep
-  integer                         :: dt_int
+  integer                         :: dt_int,ii,jj
```
Comment (jiandewang) [inline config_src/drivers/nuopc_cap/mom_cap_methods.F90]: need to add space between them

Comment (mark-a-potts) [reply config_src/drivers/nuopc_cap/mom_cap_methods.F90]: They were actually unused, so I removed them.

**Comment by mark-a-potts** [comment]
> 

Sorry about that. I squashed all the changes since merging to dev/emc into one and renamed it, but the list of commits are still showing here. I can re-do the PR with a clean push, if you would like.

**Comment by jiandewang** [comment]
> > 
> 
> Sorry about that. I squashed all the changes since merging to dev/emc into one and renamed it, but the list of commits are still showing here. I can re-do the PR with a clean push, if you would like.

if you can re-do that in a clean push, that will be great. Thanks.

**Pull request closed without merging**